fix: LEAP-1433: Remove deprecated feature flag FF_DEV_1170

Gondragos opened the pull request:
Eliminated references to the obsolete feature flag `FF_DEV_1170` (`ff_front_1170_outliner_030222_short`) across multiple files, including tests, components, and utilities. Key changes include:
- Updated relevant tests by removing the flag setup and associated logic.
- Refactored components to exclude conditional logic tied to the flag.
- Cleaned up feature-flag utility files by removing the declaration of the flag.

| **Key** | `ff_front_1170_outliner_030222_short` |
|---|---|
| **Variable Name** | `FF_DEV_1170` |
| **Ends up as** | 🟩 `True` |
| **Description** | Outliner + Details |

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
fix: LEAP-1433: Remove deprecated feature flag FF_DEV_1170

Eliminated references to the obsolete feature flag `FF_DEV_1170` (`ff_front_1170_outliner_030222_short`) across multiple files, including tests, components, and utilities.
Key changes include:
- Updated relevant tests by removing the flag setup and associated logic.
- Refactored components to exclude conditional logic tied to the flag.
- Cleaned up feature-flag utility files by removing the declaration of the flag.

Gondragos committed:
tests: Replace AtSidebar with AtOutliner in tests

Updated all test files to replace references to `AtSidebar` with `AtOutliner`. This ensures uniformity and aligns with recent architectural changes. Adjustments include replacing function calls, variable names, and related interactions across multiple test scenarios.

Gondragos committed:
refactor: remove unused AtSidebar helper

Eliminated the AtSidebar.js file as it is no longer required or referenced in the codebase.

Gondragos committed:
test: reduce region creation iterations in audio test

Reduced the number of iterations from 20 to 10 in E2E tests for audio regions. This change optimizes test execution time without compromising functionality validation. Updated region checks to match the new count.

Gondragos committed:
fix: reduce iteration count in audio decoder test

Reduced the number of iterations in the `audio-webaudio-decoder` E2E test from 20 to 10. This change ensures faster test execution while maintaining test coverage and functionality. Adjusted associated assertions to match the new iteration count.

Gondragos committed:
fix: e2e tests improvements  pt.k+1

    Integrated `disableDefaultInit` plugin for better control during page setup in tests. Updated usages of "Control" to "CommandOrControl" for cross-platform compatibility. Improved calculations and element targeting in timeline and region-based tests. Refined test logic to prevent false positives and ensure better reliability of assertions.

Gondragos committed:
fix: skip empty labels test for classifications

bmartel commented:
@yyassi-heartex and I were just talking about how it would be really amazing to clean this up, awesome 🎉

Gondragos committed:
test: Fix bunch of tests

Most of the fixes are based on closing details panel to restore original size of the image at the moment test had been written.

Gondragos committed:
fix: Fix shortcuts tests

Gondragos committed:
feat: Fix smoke tests

Added hotkey support and ARIA labels for region actions (visibility toggle and delete). Enhanced e2e smoke tests to improve usability and test coverage. Fixed missing parent check for Label components on various region types.

Gondragos committed:
test: ensure objects are ready before taxonomy actions

Added `LabelStudio.waitForObjectsReady()` to multiple test cases within `taxonomy.test.js`. This ensures that all objects are fully initialized before performing actions on the taxonomy, reducing potential race conditions.

Gondragos committed:
fix: Collapse details panel during gesture tests

Gondragos committed:
fix: add wait for objects readiness in test

Ensure objects are ready before checking annotation button behavior in the regression test. Added a `LabelStudio.waitForObjectsReady()` call to prevent potential timing issues and improve test stability.

Gondragos committed:
fix: add waits to transformer test

Gondragos committed:
fix: resolve flaky e2e tests by ensuring objects load

Fixed flaky end-to-end tests by adding `LabelStudio.waitForObjectsReady()` where necessary and addressing CI issues.
- Added `LabelStudio.waitForObjectsReady()` to ensure objects are loaded before proceeding in tests.
- Temporarily disabled a specific scenario due to size-related issues in CI by replacing `Data` with `xData`.
- Adjusted panel collapsing order in `image-ctrl-drawing.test.js` for better consistency.

Gondragos committed:
fix: replace image wait method with objects ready check

Replaced `AtImageView.waitForImage()` with `LabelStudio.waitForObjectsReady()` in regression tests.

Gondragos committed:
fix: replace waitForImage with waitForObjectsReady

Updated e2e tests to use LabelStudio.waitForObjectsReady instead of AtImageView.waitForImage for improved reliability and consistency. Affected multiple regression test files, ensuring proper initialization of objects before proceeding with tests.

Gondragos committed:
fix: ensure objects are ready before assertions

Added `LabelStudio.waitForObjectsReady` in multiple regression tests to ensure objects are fully initialized before performing actions or assertions. This change improves test stability and prevents intermittent failures caused by timing issues.

Gondragos committed:
refactor: replace deprecated test utility methods

Replaced `I.executeScript(initLabelStudio)` and `AtImageView.waitForImage` calls with standardized `LabelStudio.init` and `LabelStudio.waitForObjectsReady` methods in all tests.

Gondragos committed:
fix: return waitForAudio check

Gondragos commented on web/libs/editor/tests/e2e/tests/rich-text/rich-text-edge-cases.test.js:
```diff
@@ -2,20 +2,14 @@ const assert = require("assert");
 </View>`;
 const edgeCaseHtml = `<p>This
-is an <u><span>e</span><span>x</span><span>a</span><span>m</span><span>p</span><span>l</span><span>e</span></u><br/> of 
+is an <u><span>e</span><span>x</span><span>a</span><span>m</span><span>p</span><span>l</span><span>e</span></u><br/> of
```
```suggestion
is an <u><span>e</span><span>x</span><span>a</span><span>m</span><span>p</span><span>l</span><span>e</span></u><br/> of 
```

Commit:
I hate linters

Gondragos committed:
fix: Fix undo/redo test

Updated keybinding references from "ctrl" to platform-agnostic "CommandOrControl" for consistency across environments. Added panel collapse functionality in a specific e2e test scenario to ensure proper initial state.

Gondragos committed:
fix: Bring back skipped test

Gondragos committed:
fix: Return waitForAudio check to smoke test

Gondragos committed:
test: add missed arg

Gondragos committed:
refactor: remove unused components and icons

Gondragos committed:
Merge branch 'develop' into fb-LEAP-1433/stale-outliner-ff

# Conflicts:
#	web/libs/editor/src/assets/icons/index.tsx
#	web/libs/editor/src/components/CurrentEntity/Controls.jsx
#	web/libs/editor/src/components/CurrentEntity/HistoryActions.jsx
#	web/libs/editor/src/components/Entities/Entities.jsx
#	web/libs/editor/src/components/Entities/LabelItem.jsx
#	web/libs/editor/src/components/Entities/LabelList.jsx
#	web/libs/editor/src/components/Entities/RegionItem.jsx
#	web/libs/editor/src/components/Entities/RegionTree.jsx
#	web/libs/editor/src/components/Entities/SortMenu.jsx
#	web/libs/editor/src/components/Entity/Entity.jsx
#	web/libs/editor/src/components/Node/Node.tsx
#	web/libs/ui/src/assets/icons/collapse.svg
#	web/libs/ui/src/assets/icons/date.svg
#	web/libs/ui/src/assets/icons/expand.svg
#	web/libs/ui/src/assets/icons/invisible.svg
#	web/libs/ui/src/assets/icons/visible.svg

Gondragos committed:
refactor: remove unused imports from Node.tsx

Gondragos committed:
feat: restore expand and collapse icons

Gondragos committed:
feat: restore NodeView import

Gondragos committed:
test: fix flaky transformer test

yyassi-heartex reviewed (approved):
excited to see this get cleaned up!

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14101237319)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14101237319

Gondragos committed:
Merge branch 'develop' into fb-LEAP-1433/stale-outliner-ff

# Conflicts:
#	label_studio/core/feature_flags/stale_feature_flags.py
#	web/libs/editor/tests/e2e/fragments/AtSidebar.js
#	web/libs/editor/tests/e2e/tests/image.transformer.test.js
#	web/libs/editor/tests/e2e/tests/regression-tests/image-draw-undo.test.js
#	web/libs/editor/tests/e2e/tests/unfinished-polygons.test.js

Gondragos committed:
test: Add support for incomplete state icons in AtOutliner

Introduced methods to locate and verify visibility of incomplete state icons in region lists. Refactored existing region-related methods to improve reusability by introducing `locateRegion`. Updated tests to include assertions for incomplete state icons.

hlomzik commented on web/libs/editor/src/components/App/App.jsx:
```diff
@@ -279,56 +275,37 @@ class App extends Component {
+                  </SideTabsPanels>
                 )
+              ) : isBulkMode ? (
+                <>{mainContent}</>
```
```suggestion
                mainContent
```

hlomzik commented on web/libs/frontend-test/src/helpers/LSF/Sidebar.ts:
```diff
@@ -28,15 +25,9 @@ export const Sidebar = {
-          .get(".lsf-tree__node:not(.lsf-tree__node_type_footer) .lsf-tree-node-content-wrapper");
-      }
-
-      return this.legacySidebar.should("be.visible").get(".lsf-region-item");
```
`legacySidebar` should also be removed then

Commit:
Apply suggestion

Co-authored-by: Andrew <hlomzik@gmail.com>

hlomzik commented on web/libs/editor/tests/e2e/tests/image.selected-region.test.js:
```diff
@@ -44,40 +43,51 @@ async function testRegion(testType, toolAccelerator, I, LabelStudio, AtImageView
-  await doDrawingAction(I, { msg: `Initial ${testType}`, fromX: 150, fromY: 110, toX: 150 + 50, toY: 110 + 50 });
+  await doDrawingAction(I, {
+    msg: `Initial ${testType}`,
+    fromX: 319 + 150,
```
would be awesome to see these numbers as consts to understand the meaning behind them

hlomzik reviewed (approved):
That's a titanic job, thank you! Small observations, but minors.

Gondragos committed:
fix: correct visibility check for incomplete regions

Updated the visibility assertion logic in `seeIncompleteRegion` to check for the presence of the incomplete state icon instead of its absence. Removed unused legacy sidebar reference in the `Sidebar` helper. Introduced `OUTLINER_PANEL_WIDTH` constant to replace hardcoded values in image region selection tests.

Gondragos committed:
Merge remote-tracking branch 'origin/fb-LEAP-1433/stale-outliner-ff' into fb-LEAP-1433/stale-outliner-ff

Gondragos committed:
fix: ensure panel is collapsed before drawing

Collapse the Details Panel to avoid interfering with region drawings during the test. This adjustment improves the reliability and accuracy of the image control drawing regression test.

Gondragos committed:
fix: try to fix

Gondragos committed:
chore: add region bounding box retrieval for tests

Gondragos committed:
test: fix region comparison in image transformer test

Gondragos committed:
test: add stage bbox recalculation

Gondragos committed:
test: experimental fix

Gondragos committed:
fix: remove wrong test step

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14265211737)
> Successfully merged:  create mode 100644 web/libs/ui/src/shadcn.ts

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1433/stale-outliner-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14265211737

Gondragos committed:
fix: fix waiting conditions

Gondragos committed:
Merge remote-tracking branch 'origin/fb-LEAP-1433/stale-outliner-ff' into fb-LEAP-1433/stale-outliner-ff

Gondragos committed:
fix: fix waiting conditions

Gondragos committed:
experimental fix

Gondragos committed:
fix: Correct inconsistent indentation in e2e tests

Gondragos committed:
refactor: apply different aproach of waiting

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14311563556)
> Successfully merged:  create mode 100644 label_studio/users/product_tours/configs/show_ask_ai.yml

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1433/stale-outliner-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14311563556

Gondragos committed:
fix: add delay to setZoom in AtImageView tests

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14333253799)
> Successfully merged:  create mode 100644 docs/themes/v2/source/images/storages/s3-trust-policy.png

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1433/stale-outliner-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14333253799

Gondragos commented on web/libs/editor/tests/e2e/fragments/AtImageView.js:
```diff
@@ -106,6 +110,7 @@ module.exports = {
 
   setZoom(scale, x, y) {
     I.executeScript(Helpers.setZoom, [scale, x, y]);
+    I.wait(1);
```
```suggestion
    I.wait(2);
```
It looks like it may be not enough time to wait

Commit:
Change waiting time

Gondragos committed:
fix: correct zoom handling in image transformer tests

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14368999730)
> Successfully merged:  create mode 100644 web/libs/storybook/addons/theme-toggle/theme-toggle.tsx

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1433/stale-outliner-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14368999730
